svitifouka.go

```diff
@@ -107,8 +91,7 @@ func getEntsoeData() map[string]int {
 	timeInterval := getPastHourInterval()
 	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
-		log.Print(err)
-		os.Exit(1)
+		log.Fatal(err)
```

Comment:
Nebudu to opakovat na dalších místech, ale princip je všude stejný: errory by neměly ukončovat jednotlivé funkce takhle fatálně - nejen že se to blbě odchytává, ale hlavně to jde proti principu Go error handlingu. Konkrétně tahle funkce by místo `string` měla vracet `(string, error)` a v tuto chvíli bys vrátil prázdný string a ten error. Je pak na callerovi, aby se rozhodl, co s tím errorem provede (zpravidla jej pošle opět svému "rodiči" a handluje se pak úplně navrchu).

Reply:
Rozumím, díky moc!

Ještě z druhé strany, vstupu. U téhle funkce to zrovna není relevantní, ale jak bych měl pracovat s proměnnými, které jsou definované v globálním prostoru (tady třeba `resList` nebo `runeMap`) ve funkcích? Měl bych je mít jako parametry a potom argumenty ve funkcích nebo u takto jednoduchého programu je to jedno a může to zůstat jak je?

Reply:
Zrovna tady ty tvoje mapy fungujou spíš jako konstanty, takže je to v jednoduchém programu IMO v pohodě (ale nějakého puristu by to vytriggerovat mohlo). V ničem produkčním bych to takhle nedělal, pořád je to globální modifikovatelný stav, což je vždycky zdroj nepříjemných chyb.

Reply:
Děkuji moc za tuto i ostatní zpětnou vazbu!